WooCommerce/Classes/ViewRelated/Coupons/CouponDetails/CouponDetailsViewModel.swift

```diff
@@ -95,7 +98,7 @@ final class CouponDetailsViewModel: ObservableObject {
         hasErrorLoadingAmount = false
         // Get "ancient" date to fetch all possible reports
         let startDate = Date(timeIntervalSince1970: 1)
-        let action = CouponAction.loadCouponReport(siteID: coupon.siteID, couponID: coupon.couponID, startDate: startDate) { [weak self] result in
+        let action = CouponAction.loadCouponReport(siteID: siteID, couponID: coupon.couponID, startDate: startDate) { [weak self] result in
```

Comment:
I believe the site IDs have to always match since we're fetching coupons by site ID. The reason why I extracted site ID to a separate property is because I need to access it from the view too 😅